### DISCUSSION TITLE
Add a `--wait` option to some `niri msg` actions.

### DISCUSSION THREAD

**Discussion by minego** [YaLTeR/niri#471] [Ideas]
I've been using `niri msg --json focused-window` to get the title of the active window, to display in a bar. This works just fine, but there is no way to know when it has changed, so I am currently running it every second.

This feels rather wasteful, and it means that there is a short delay before the title actually updates.

I'd love to see a new option added to `niri msg` that allows waiting for a change before it returns. That would work perfectly with `ags` since it can run async commands and wait for the result.

**Comment by YaLTeR** [comment]
This will be implementable properly without races using https://github.com/YaLTeR/niri/pull/453

**Comment by Okanda** [comment]
Personnaly, I choose the window I want to know the name, launch fuzzel and then run "kgx niri msg focused-window".

**Reply by minego** [reply]
> Personnaly, I choose the window I want to know the name, launch fuzzel and then run "kgx niri msg focused-window".

Yes, and that works perfectly for manual use. I'm calling it from a bar, so I can show the title in my bar. I'm trying to avoid the delay and wasted CPU cycles running it once a second
